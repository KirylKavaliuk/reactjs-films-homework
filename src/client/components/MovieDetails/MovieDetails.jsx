import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Button from 'components/Button/Button';
import Rating from 'components/Rating/Rating';
import FilmHeader from 'components/FilmHeader/FilmHeader';
import Search from 'components/Search/Search';
import Description from 'components/Description/Description';

import withConditionalRendering from 'utils/withConditionalRendering';

import styles from './MovieDetails.scss';

class MovieDetails extends Component {
  state = {
    descriptionOpen: false,
  };

  toggleDescriptionHandler = () => {
    this.setState({ descriptionOpen: !this.state.descriptionOpen });
  }

  render() {
    return (
      <div className={ styles.movieDetails }>
        <a
          href="#"
          className={ styles.logo }
        >Films</a>

        <Search
          onChange={ this.props.changeSearch }
          value={ this.props.searchValue }
        />

        <FilmHeader
          name={ this.props.movie.name }
          genres={ this.props.movie.genres }
          duration={ this.props.movie.duration }
        />

        <Rating value={ this.props.movie.rating }/>

        <div className={ styles.buttonsMore }>
          <Description
            open={ this.state.descriptionOpen }
            text={ this.props.movie.description }
          />
          <Button label='Watch Now'/>
          <Button
            label='View Info'
            transparent={ true }
            active={ this.state.descriptionOpen }
            onClick={ this.toggleDescriptionHandler }
          />
        </div>
      </div>
    );
  }
}

MovieDetails.defaultProps = {
  movie: null,
};

MovieDetails.propTypes = {
  movie: PropTypes.shape({
    name: PropTypes.string,
    genres: PropTypes.array,
    duration: PropTypes.number,
    rating: PropTypes.number,
    description: PropTypes.string,
  }),
};

export default withConditionalRendering(MovieDetails, ['movie']);
