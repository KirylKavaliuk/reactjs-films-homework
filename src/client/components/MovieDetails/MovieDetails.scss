@import 'styles/palette';

.movie-details {
  display: grid;
  width: 100%;
  min-height: 70vh;
  background: url('images/details-background.jpg') no-repeat center center;
  background-size: cover;
  box-shadow:
    inset 0 -150px 150px rgba(0, 0, 0, 0.8),
    inset 0 50px 100px rgba(0, 0, 0, 0.8);
  grid-template-columns: 80px minmax(150px, 70%) minmax(150px, 30%) 80px;
  grid-template-rows: 80px auto 80px;
  grid-template-areas:
    ". logo search ."
    ". film-header . ."
    ". rating buttons .";
  color: $white;
}

.logo {
  text-transform: uppercase;
  font-size: 40px;
  grid-area: logo;
  align-self: center;
}

.buttons-more {
  position: relative;
  display: flex;
  width: 300px;
  justify-content: space-between;
  grid-area: buttons;
  align-self: center;
  justify-self: end;
}

@media only screen and (max-width: 1000px) {
  .movie-details {
    min-height: 80vh;
    grid-template-columns: 20px minmax(150px, 60%) minmax(150px, 40%) 20px;
    grid-template-rows: 80px auto 80px;
  }
}

@media only screen and (max-width: 750px) {
  .movie-details {
    grid-template-columns: 10px auto 10px;
    grid-template-rows: 0 80px auto 60px 60px;
    grid-template-areas:
      ". . ."
      ". search ."
      ". film-header ."
      ". rating ."
      ". buttons .";
  }

  .buttons-more {
    width: 100%;
    justify-content: space-around;
  }

  .logo {
    display: none;
  }
}

@media only screen and (max-width: 450px) {
  .movie-details {
    min-height: 90vh;
    grid-template-rows: 0 80px auto 60px 110px;
  }

  .buttons-more {
    flex-direction: column-reverse;
    width: 100%;
    height: 100%;
    align-items: stretch;
    justify-content: space-around;
  }
}
