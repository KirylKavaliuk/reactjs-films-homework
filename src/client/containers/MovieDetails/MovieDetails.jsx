import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Button from 'Components/Button/Button';
import Rating from 'Components/Rating/Rating';
import FilmHeader from 'Components/FilmHeader/FilmHeader';
import Search from 'Components/Search/Search';

import styles from './MovieDetails.scss';

export default class MovieDetails extends Component {
  constructor(props) {
    super(props);

    this.state = {
      search: '',
    };

    this.onChangeSearchHandler = this.onChangeSearchHandler.bind(this);
  }

  onChangeSearchHandler(event) {
    this.setState({ search: event.target.value });
  }

  render() {
    if (this.props.movie) {
      return (
        <section className={ styles.movieDetails }>
          <div className={ styles.gradientBottom }></div>
          <div className={ styles.gradientTop }></div>
          <div className={ styles.content}>
            <div className={ styles.top }>
              <div className={ styles.logo }>
                Films
              </div>
              <Search
                value={ this.state.search }
                onChange={ this.onChangeSearchHandler }
              />
            </div>
            <div className={ styles.bottom }>
              <div className={ styles.briefInfo }>
                <FilmHeader
                  name={ this.props.movie && this.props.movie.name }
                  genres={ this.props.movie.genres }
                  duration={ this.props.movie.duration }
                />

                <Rating
                  value={ this.props.movie.rating }
                />
              </div>
              <div className={ styles.moreButtons }>
                <Button
                  label='Watch Now'
                />

                <Button
                  label='View Info'
                  transparent
                />
              </div>
            </div>
          </div>
        </section>
      );
    }
    return null;
  }
}

MovieDetails.defaultProps = {
  movie: null,
};

MovieDetails.propTypes = {
  movie: PropTypes.shape({
    name: PropTypes.string,
    genres: PropTypes.array,
    duration: PropTypes.number,
    rating: PropTypes.number,
    description: PropTypes.string,
  }),
};
