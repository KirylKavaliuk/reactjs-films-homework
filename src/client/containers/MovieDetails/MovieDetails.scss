.movie-details {
  background: url('Images/details-background.jpg') no-repeat center center;
  background-size: cover;
  min-height: 70vh;
  width: 100%;
  position: relative;
}

.gradient-bottom {
  position: absolute;
  width: 100%;
  height: 100%;
  background: linear-gradient(0deg, rgba(0, 0, 0, 0.9) 0%, rgba(0, 0, 0, 0.89) 1%, rgba(155, 155, 155, 0) 62%, rgba(252, 252, 252, 0) 100%);
}

.gradient-top {
  position: absolute;
  width: 100%;
  height: 100%;
  background: linear-gradient(0deg, rgba(0, 0, 0, 0) 0%, rgba(0, 0, 0, 0) 65%, rgba(0, 0, 0, 0.8) 100%);
}

.content {
  position: absolute;
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;

  .top {
    display: flex;
    justify-content: space-between;
    width: 100%;
    padding: 20px 80px;
    box-sizing: border-box;
  }

  .bottom {
    display: flex;
    justify-content: space-between;
    align-items: flex-end;
    width: 100%;
    padding: 20px 80px;
    box-sizing: border-box;

    .more-buttons {
      display: flex;
      justify-content: space-between;
      width: 100%;
      max-width: 300px;
      padding-bottom: 20px;
      box-sizing: border-box;
    }
  }

  .logo {
    font-size: 36px;
    text-transform: uppercase;
  }
}
