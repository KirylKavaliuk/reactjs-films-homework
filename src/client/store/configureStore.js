import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import reducers from './reducers';
/* istanbul ignore next line */
const composeEnhancers = typeof window === 'object'
  // eslint-disable-next-line
  && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
  && process.env.NODE_ENV !== 'production'
  // eslint-disable-next-line
  ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
  : compose;

export default (initialState = {}) => {
  const store = createStore(
    reducers,
    initialState,
    composeEnhancers(applyMiddleware(thunk)),
  );

  /* istanbul ignore next line */
  if (module.hot && process.env.NODE_ENV !== 'production') {
    /* istanbul ignore next line */
    module.hot.accept('./reducers/index', () => {
      store.replaceReducer(reducers);
    });
  }

  return store;
};
