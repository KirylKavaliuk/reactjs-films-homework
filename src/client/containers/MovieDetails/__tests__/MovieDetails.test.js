import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import ShallowRenderer from 'react-test-renderer/shallow';
import TestRenderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'store/configureStore';
import MovieDetails from '../MovieDetails';

const movie = {
  id: 12344455,
  name: 'Film',
  genres: [
    { id: 1, name: 'Adventure' },
    { id: 2, name: 'Drama' },
    { id: 3, name: 'Family' },
    { id: 4, name: 'Fantasy' },
  ],
  runtime: 237,
  vote_average: 2.54,
  overview: 'Description text description text description text description text description text description text',
};

const movieWithoutGenres = {
  name: 'Film',
  genres: null,
  runtime: 237,
  vote_average: 2.54,
  overview: 'Description text description text description text description text description text description text',
};

it('MovieDetails.test', () => {
  const tree = TestRenderer.create(
    <MemoryRouter>
    <Provider store={ configureStore({
      movieDetails: movie,
    }) }>
      <MovieDetails movie={ movie }/>
    </Provider>
    </MemoryRouter>,
  );

  const { root } = tree;

  let search = root.findByType('input').props;
  search.onChange({ target: { value: 'test search value' } });
  expect(tree.toJSON()).toMatchSnapshot();
  search = root.findByType('input').props;
  expect(search.value).toBe('test search value');

  const descButton = root.findByProps({ label: 'View Info' }).props;
  descButton.onClick();
  expect(tree.toJSON()).toMatchSnapshot();
  descButton.onClick();
  expect(tree.toJSON()).toMatchSnapshot();

  const tree2 = TestRenderer.create(
    <MemoryRouter>
    <Provider store={ configureStore() }>
      <MovieDetails movie={ movieWithoutGenres }/>
    </Provider>
    </MemoryRouter>,
  );

  expect(tree2.toJSON()).toMatchSnapshot();
});
