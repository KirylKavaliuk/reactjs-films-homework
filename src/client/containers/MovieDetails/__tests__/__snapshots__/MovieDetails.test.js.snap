// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails.test 1`] = `
<div
  className="MovieDetails__movie-details___h0XRr"
  id="movie-details"
  style={
    Object {
      "backgroundImage": "url(https://image.tmdb.org/t/p/w1280undefined)",
    }
  }
>
  <a
    className="Link__link___-IU0S MovieDetails__logo___1eswn"
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Films
  </a>
  <form
    className="Search__search___1reG0"
    name="search"
  >
    <input
      className="Search__search-field___31kXA"
      onChange={[Function]}
      placeholder="Search..."
      type="search"
      value="test search value"
    />
    <button
      className="Search__search-button___1ljeO"
      onClick={[Function]}
    >
      <svg
        className=""
        viewBox="0 0 483.083 483.083"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M332.74 315.35c30.883-33.433 50.15-78.2 50.15-127.5C382.89 84.433 298.74 0 195.04 0S7.19 84.433 7.19 187.85 91.34 375.7 195.04 375.7c42.217 0 81.033-13.883 112.483-37.4l139.683 139.683c3.4 3.4 7.65 5.1 11.9 5.1s8.783-1.7 11.9-5.1c6.517-6.517 6.517-17.283 0-24.083L332.74 315.35zM41.19 187.85C41.19 103.133 110.04 34 195.04 34c84.717 0 153.85 68.85 153.85 153.85S280.04 341.7 195.04 341.7 41.19 272.567 41.19 187.85z"
        />
      </svg>
    </button>
  </form>
  <div
    className="FilmHeader__film-header___1bjD8"
  >
    <h2
      className="FilmHeader__name___12D1E"
    >
      
    </h2>
    <div
      className="FilmHeader__info___2BQ5Y"
    >
      <span
        className="FilmHeader__genre___2QZId"
      >
        Adventure
      </span>
      <span
        className="FilmHeader__genre___2QZId"
      >
        Drama
      </span>
      <span
        className="FilmHeader__genre___2QZId"
      >
        Family
      </span>
      <span
        className="FilmHeader__duration___1HHqc"
      >
        3 h
         
        57 m
      </span>
    </div>
  </div>
  <div
    className="Rating__rating___3f4Qx"
  >
    <div
      className="Rating__star___3Zoo4 Rating__active___BzsSn"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    />
    <div
      className="Rating__star___3Zoo4"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    />
    <div
      className="Rating__star___3Zoo4"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    />
    <div
      className="Rating__star___3Zoo4"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    />
    <div
      className="Rating__star___3Zoo4"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    />
    <div
      className="Rating__rating-number-wrap___dhLPE"
    >
      <div
        className="Rating__rating-number___13U9X"
      >
        1.27
      </div>
    </div>
  </div>
  <div
    className="MovieDetails__buttons-more___8NsJC"
  >
    <div
      className="Description__description___1aDcr"
    >
      Description text description text description text description text description text description text
    </div>
    <a
      className="Link__link___-IU0S"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <button
        className="Button__button___1nbn1"
        onClick={null}
      >
        Watch Now
      </button>
    </a>
    <button
      className="Button__button___1nbn1 Button__transparent___3tNuU"
      onClick={[Function]}
    >
      View Info
    </button>
  </div>
</div>
`;

exports[`MovieDetails.test 2`] = `
<div
  className="MovieDetails__movie-details___h0XRr"
  id="movie-details"
  style={
    Object {
      "backgroundImage": "url(https://image.tmdb.org/t/p/w1280undefined)",
    }
  }
>
  <a
    className="Link__link___-IU0S MovieDetails__logo___1eswn"
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Films
  </a>
  <form
    className="Search__search___1reG0"
    name="search"
  >
    <input
      className="Search__search-field___31kXA"
      onChange={[Function]}
      placeholder="Search..."
      type="search"
      value="test search value"
    />
    <button
      className="Search__search-button___1ljeO"
      onClick={[Function]}
    >
      <svg
        className=""
        viewBox="0 0 483.083 483.083"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M332.74 315.35c30.883-33.433 50.15-78.2 50.15-127.5C382.89 84.433 298.74 0 195.04 0S7.19 84.433 7.19 187.85 91.34 375.7 195.04 375.7c42.217 0 81.033-13.883 112.483-37.4l139.683 139.683c3.4 3.4 7.65 5.1 11.9 5.1s8.783-1.7 11.9-5.1c6.517-6.517 6.517-17.283 0-24.083L332.74 315.35zM41.19 187.85C41.19 103.133 110.04 34 195.04 34c84.717 0 153.85 68.85 153.85 153.85S280.04 341.7 195.04 341.7 41.19 272.567 41.19 187.85z"
        />
      </svg>
    </button>
  </form>
  <div
    className="FilmHeader__film-header___1bjD8"
  >
    <h2
      className="FilmHeader__name___12D1E"
    >
      
    </h2>
    <div
      className="FilmHeader__info___2BQ5Y"
    >
      <span
        className="FilmHeader__genre___2QZId"
      >
        Adventure
      </span>
      <span
        className="FilmHeader__genre___2QZId"
      >
        Drama
      </span>
      <span
        className="FilmHeader__genre___2QZId"
      >
        Family
      </span>
      <span
        className="FilmHeader__duration___1HHqc"
      >
        3 h
         
        57 m
      </span>
    </div>
  </div>
  <div
    className="Rating__rating___3f4Qx"
  >
    <div
      className="Rating__star___3Zoo4 Rating__active___BzsSn"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    />
    <div
      className="Rating__star___3Zoo4"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    />
    <div
      className="Rating__star___3Zoo4"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    />
    <div
      className="Rating__star___3Zoo4"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    />
    <div
      className="Rating__star___3Zoo4"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    />
    <div
      className="Rating__rating-number-wrap___dhLPE"
    >
      <div
        className="Rating__rating-number___13U9X"
      >
        1.27
      </div>
    </div>
  </div>
  <div
    className="MovieDetails__buttons-more___8NsJC"
  >
    <div
      className="Description__description___1aDcr Description__open___3dk14"
    >
      Description text description text description text description text description text description text
    </div>
    <a
      className="Link__link___-IU0S"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <button
        className="Button__button___1nbn1"
        onClick={null}
      >
        Watch Now
      </button>
    </a>
    <button
      className="Button__button___1nbn1 Button__transparent___3tNuU Button__active___2cdo3"
      onClick={[Function]}
    >
      View Info
    </button>
  </div>
</div>
`;

exports[`MovieDetails.test 3`] = `
<div
  className="MovieDetails__movie-details___h0XRr"
  id="movie-details"
  style={
    Object {
      "backgroundImage": "url(https://image.tmdb.org/t/p/w1280undefined)",
    }
  }
>
  <a
    className="Link__link___-IU0S MovieDetails__logo___1eswn"
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Films
  </a>
  <form
    className="Search__search___1reG0"
    name="search"
  >
    <input
      className="Search__search-field___31kXA"
      onChange={[Function]}
      placeholder="Search..."
      type="search"
      value="test search value"
    />
    <button
      className="Search__search-button___1ljeO"
      onClick={[Function]}
    >
      <svg
        className=""
        viewBox="0 0 483.083 483.083"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M332.74 315.35c30.883-33.433 50.15-78.2 50.15-127.5C382.89 84.433 298.74 0 195.04 0S7.19 84.433 7.19 187.85 91.34 375.7 195.04 375.7c42.217 0 81.033-13.883 112.483-37.4l139.683 139.683c3.4 3.4 7.65 5.1 11.9 5.1s8.783-1.7 11.9-5.1c6.517-6.517 6.517-17.283 0-24.083L332.74 315.35zM41.19 187.85C41.19 103.133 110.04 34 195.04 34c84.717 0 153.85 68.85 153.85 153.85S280.04 341.7 195.04 341.7 41.19 272.567 41.19 187.85z"
        />
      </svg>
    </button>
  </form>
  <div
    className="FilmHeader__film-header___1bjD8"
  >
    <h2
      className="FilmHeader__name___12D1E"
    >
      
    </h2>
    <div
      className="FilmHeader__info___2BQ5Y"
    >
      <span
        className="FilmHeader__genre___2QZId"
      >
        Adventure
      </span>
      <span
        className="FilmHeader__genre___2QZId"
      >
        Drama
      </span>
      <span
        className="FilmHeader__genre___2QZId"
      >
        Family
      </span>
      <span
        className="FilmHeader__duration___1HHqc"
      >
        3 h
         
        57 m
      </span>
    </div>
  </div>
  <div
    className="Rating__rating___3f4Qx"
  >
    <div
      className="Rating__star___3Zoo4 Rating__active___BzsSn"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    />
    <div
      className="Rating__star___3Zoo4"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    />
    <div
      className="Rating__star___3Zoo4"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    />
    <div
      className="Rating__star___3Zoo4"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    />
    <div
      className="Rating__star___3Zoo4"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    />
    <div
      className="Rating__rating-number-wrap___dhLPE"
    >
      <div
        className="Rating__rating-number___13U9X"
      >
        1.27
      </div>
    </div>
  </div>
  <div
    className="MovieDetails__buttons-more___8NsJC"
  >
    <div
      className="Description__description___1aDcr"
    >
      Description text description text description text description text description text description text
    </div>
    <a
      className="Link__link___-IU0S"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <button
        className="Button__button___1nbn1"
        onClick={null}
      >
        Watch Now
      </button>
    </a>
    <button
      className="Button__button___1nbn1 Button__transparent___3tNuU"
      onClick={[Function]}
    >
      View Info
    </button>
  </div>
</div>
`;

exports[`MovieDetails.test 4`] = `
<div
  className="MovieDetails__movie-details___h0XRr"
  id="movie-details"
  style={
    Object {
      "backgroundImage": "",
    }
  }
>
  <a
    className="Link__link___-IU0S MovieDetails__logo___1eswn"
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Films
  </a>
  <form
    className="Search__search___1reG0"
    name="search"
  >
    <input
      className="Search__search-field___31kXA"
      onChange={[Function]}
      placeholder="Search..."
      type="search"
      value=""
    />
    <button
      className="Search__search-button___1ljeO"
      onClick={[Function]}
    >
      <svg
        className=""
        viewBox="0 0 483.083 483.083"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M332.74 315.35c30.883-33.433 50.15-78.2 50.15-127.5C382.89 84.433 298.74 0 195.04 0S7.19 84.433 7.19 187.85 91.34 375.7 195.04 375.7c42.217 0 81.033-13.883 112.483-37.4l139.683 139.683c3.4 3.4 7.65 5.1 11.9 5.1s8.783-1.7 11.9-5.1c6.517-6.517 6.517-17.283 0-24.083L332.74 315.35zM41.19 187.85C41.19 103.133 110.04 34 195.04 34c84.717 0 153.85 68.85 153.85 153.85S280.04 341.7 195.04 341.7 41.19 272.567 41.19 187.85z"
        />
      </svg>
    </button>
  </form>
  <div
    className="FilmHeader__film-header___1bjD8"
  >
    <h2
      className="FilmHeader__name___12D1E"
    >
      
    </h2>
    <div
      className="FilmHeader__info___2BQ5Y"
    >
      0
    </div>
  </div>
  <div
    className="Rating__rating___3f4Qx"
  >
    <div
      className="Rating__star___3Zoo4"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    />
    <div
      className="Rating__star___3Zoo4"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    />
    <div
      className="Rating__star___3Zoo4"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    />
    <div
      className="Rating__star___3Zoo4"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    />
    <div
      className="Rating__star___3Zoo4"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    />
    <div
      className="Rating__rating-number-wrap___dhLPE"
    >
      <div
        className="Rating__rating-number___13U9X"
      >
        0.00
      </div>
    </div>
  </div>
  <div
    className="MovieDetails__buttons-more___8NsJC"
  >
    <a
      className="Link__link___-IU0S"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <button
        className="Button__button___1nbn1"
        onClick={null}
      >
        Watch Now
      </button>
    </a>
    <button
      className="Button__button___1nbn1 Button__transparent___3tNuU"
      onClick={[Function]}
    >
      View Info
    </button>
  </div>
</div>
`;
